hera_cal/lstbin.py

```diff
@@ -0,0 +1,893 @@
+import copy
+import argparse
+import functools
+import numpy as np
+from pyuvdata import UVCal, UVData
+from pyuvdata import utils as uvutils
+from hera_cal import omni, utils, firstcal, cal_formats, redcal, abscal
+from hera_cal.datacontainer import DataContainer
+from scipy import signal
+from scipy import interpolate
+from scipy import spatial
+import itertools
+import operator
+from astropy import stats as astats
+import gc as garbage_collector
+import datetime
+
+
+def lst_bin(data_list, lst_list, lst_grid=None, wgts_list=None, lst_init=np.pi, dlst=None,
+            lst_low=None, lst_hi=None, wrap_point=2*np.pi, atol=1e-8, median=False,
+            sigma_clip=False, sigma=2.0, return_no_avg=False, verbose=True):
+    """
+    Bin data in Local Sidereal Time (LST) onto an LST grid. An LST grid
+    is defined as an array of points increasing in Local Sidereal Time, with each point marking
+    the center of the LST bin.
+
+    Parameters:
+    -----------
+    data_list : type=list, list of DataContainer dictionaries holding complex visibility data
+
+    lst_list : type=list, list of ndarrays holding LST stamps of each data dictionary in data_list
+    
+    lst_grid : type=ndarray, uniform 1D ndarray of LST grid to bin data into (marking center of bin).
+                        Note: ideally this should not start at 0, but at 0 + dlst / 2.
+
+    wgts_list : type=list, list of DataContainer dictionaries holding weights for each dict in data_list
+
+    lst_init : type=float, starting LST to initialize lst_grid at (if not fed). This marks the left-edge
+                        of the first LST bin.
+
+    dlst : type=float, delta-LST spacing for lst_grid.
+
+    lst_low : type=float, lower bound in lst_grid centers used for LST binning
+
+    lst_hi : type=float, upper bound in lst_grid centers used for LST binning
+
+    wrap_point : type=float, end point of a single day in LST
+
+    atol : type=float, absolute tolerance for comparing floats in lst_list to floats in lst_grid
+
+    median : type=boolean, if True use median for LST binning, else use mean
+
+    sigma_clip : type=boolean, if True, perform a sigma clipping algorithm of the LST bins
+        on the real and imag components separately. Warning: This considerably slows down the code.
+
+    sigma : type=float, input standard deviation to use for sigma clipping algorithm.
+
+    return_no_avg : type=boolean, if True, return binned but un-averaged data and wgts.
+
+    Output: (data_avg, wgts_avg, data_std, lst_bins, data_num)
+    -------
+    data_avg : dictionary of data having averaged the LST bins
+
+    wgts_avg : dictionary of data weights averaged in LST bins
+    
+    data_std : dictionary of data with real component holding std along real axis
+        and imag component holding std along imag axis
+
+    lst_bins : ndarray containing final lst grid of data
+
+    data_num : dictionary containing the number of data points averaged in each LST bin.
```

Comment:
can we call this data_count instead? data_num doesn't convey count values to me. 
I would also rearrange outputs lst_bins comes out before data_std.